windows/src/wmi.cc

```diff
@@ -43,9 +43,17 @@ namespace leatherman { namespace windows {
             if (hres == RPC_E_CHANGED_MODE) {
                 LOG_DEBUG("using prior COM concurrency model");
             } else {
-                throw wmi_exception(format_hresult(_("failed to initialize COM library"), hres));
+                // Retry with multi-threaded, in case we're on Nano Server
+                hres = CoInitializeEx(0, COINIT_MULTITHREADED);
+                if (FAILED(hres)) {
+                    throw wmi_exception(format_hresult(_("failed to initialize COM library"), hres));
+                } else {
+                    LOG_DEBUG("COM single-threaded apartment not supported, using multi-threaded");
+                }
             }
-        } else {
+        }
+
+        if (SUCCEEDED(hres)) {
```

Comment:
@MikaelSmith The nested logic here has gotten a little tricky, but I think this logic is OK, and this `SUCCEEDED` is indeed necessary.

`S_FALSE` is returned when COM is already initialized, but is a `SUCCEEDED` status, and must be paired with `CoUninitialize`.  `RPC_E_CHANGED_MODE` however, is a `FAILED` status, and can simply be ignored, without any calls to `CoUninitialize`.

Upon the call to retry as MTA, this thread can bail with a `wmi_exception` on failure, or will simply log a message in either of 2 success cases:

* `S_FALSE`, which means it's the second to init -- and must call `CoUninitialize`
* `S_OK`, which means it's the first to init -- and must call `CoUninitialize`

Overall 👍 though it might be better to make the code a little more self-evident, since  most maintainers won't be as familiar with the rules.

https://msdn.microsoft.com/en-us/library/windows/desktop/ms695279(v=vs.85).aspx
https://msdn.microsoft.com/en-us/library/windows/desktop/ff485842(v=vs.85).aspx